integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonExplainCommand.scala

```diff
@@ -42,16 +43,37 @@ case class CarbonExplainCommand(
     if (explainCommand.logicalPlan.isStreaming || isCommand) {
       explainCommand.run(sparkSession)
     } else {
-      collectProfiler(sparkSession) ++ explainCommand.run(sparkSession)
+      CarbonExplainCommand.collectProfiler(explainCommand, sparkSession) ++
+      explainCommand.run(sparkSession)
     }
   }
 
-  private def collectProfiler(sparkSession: SparkSession): Seq[Row] = {
+  override protected def opName: String = "EXPLAIN"
+}
+
+case class CarbonInternalExplainCommand(
```

Comment:
Is this required?

Reply:
yes, CarbonExtensions will use this command instead of CarbonExplainCommand